### PR TITLE
backport: Merge bitcoin#18452, 19111, 19104

### PR DESCRIPTION
**Pull request by vijaydasmp**
## How Has This Been Tested?
CI passing

## Breaking Changes
None

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Comment by vijaydasmp** [comment]
dependent upon bitcoin#18814, will proceed after merging https://github.com/dashpay/dash/pull/5152

**Comment by knst** [comment]
> dependent upon https://github.com/bitcoin/bitcoin/pull/18814, will proceed after merging https://github.com/dashpay/dash/pull/5152

@vijaydasmp 5152 is merged, please, continue.

File: src/rpc/server.cpp [hunk 1]
```diff
@@ -18,11 +18,15 @@
 
 #include <algorithm>
 #include <atomic>
+#include <cassert>
```
Comment (UdjinM6) [inline src/rpc/server.cpp:21]: 18452: missing changes in `src/node/interfaces.cpp` (`src/interfaces/node.cpp` in bitcoin)

File: src/qt/bitcoin.cpp [hunk 2]
```diff
@@ -74,6 +74,23 @@ Q_DECLARE_METATYPE(CAmount)
 Q_DECLARE_METATYPE(SynchronizationState)
 Q_DECLARE_METATYPE(uint256)
 
+static void RegisterMetaTypes()
```
Comment (UdjinM6) [inline src/qt/bitcoin.cpp:77]: 19104: missing changes in `BitcoinApplication` ctor

File: src/qt/bitcoin.cpp [hunk 3]
```diff
@@ -74,6 +74,23 @@ Q_DECLARE_METATYPE(CAmount)
...
+    qRegisterMetaType<WalletModel*>();
+  #endif
+    // Register typedefs (see http://qt-project.org/doc/qt-5/qmetatype.html#qRegisterMetaType)
+    // IMPORTANT: if CAmount is no longer a typedef use the normal variant above (see https://doc.qt.io/qt-5/qmetatype.html#qRegisterMetaType-1)
+    qRegisterMetaType<CAmount>("CAmount");
+    qRegisterMetaType<size_t>("size_t");
+
+    qRegisterMetaType<std::function<void()>>("std::function<void()>");
+    qRegisterMetaType<QMessageBox::Icon>("QMessageBox::Icon");
+}
```
Comment (UdjinM6) [inline src/qt/bitcoin.cpp:92]: 19104: missed one line https://github.com/dashpay/dash/pull/5737/commits/7d4fd557d7ac3ed29c695d2b3f040dccb536a3a4#diff-2e949323a5be6fd47f7af05ba778bbd8f0c11d9530b95de9e21690004d9ee40cL559
```suggestion
   qRegisterMetaType<interfaces::BlockAndHeaderTipInfo>("interfaces::BlockAndHeaderTipInfo");
}
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/qt/bitcoin.cpp (hunk 3 above)
Comment (vijaydasmp) [reply src/qt/bitcoin.cpp:93]: This conversation got marked resolved automatically after clicking ["commit suggestion."  Please also have a look this line change

File: src/rpc/server.cpp [hunk 4]
```diff
@@ -325,17 +329,25 @@ void StartRPC()
...
+    });
 }
 
 void StopRPC()
 {
-    LogPrint(BCLog::RPC, "Stopping RPC\n");
-    WITH_LOCK(g_deadline_timers_mutex, deadlineTimers.clear());
-    DeleteAuthCookie();
-    g_rpcSignals.Stopped();
+    static std::once_flag g_rpc_stop_flag;
```
Comment (UdjinM6) [inline src/rpc/server.cpp:341]: 18452: this line should be added in 19111, not in 18542

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — DISMISSED** [review]
utACK

**Review by PastaPastaPasta — COMMENTED** [review]
utACK df934c8eda307dbe837ae18eadb59148e9289656

**Review by PastaPastaPasta — APPROVED** [review]
re-utACK df934c8eda307dbe837ae18eadb59148e9289656

**Pull request merged**